src/audio_buffer.rs

```diff
@@ -0,0 +1,482 @@
+use std::rc::Rc;
+use std::sync::Arc;
+
+use crate::SampleRate;
+
+// @todo - put it there, but should leave in `BaseAudioContext`
+// let's just go with Wav files for now
+//
+// @note - what about using https://github.com/pdeljanov/Symphonia ?
+// seems quite complete and efficient
+// @note - should also be async, but that's not a big deal neither for now
+pub fn decode_audio_data(file: std::fs::File) -> AudioBuffer {
```

Comment:
Later (in a different PR) we should not take a file but a generic `R: Read` as input!

Reply:
sounds good as it would allow to get files from the network I guess (w/ Rust equivalent of XHR?)